src/main/java/org/example/game2048/movements/Movement.java

```diff
@@ -0,0 +1,172 @@
+package org.example.game2048.movements;
+import org.example.game2048.point.Point;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class Movement {
+
+    private static List<Point> TEMPORARY_LIST;
+    private static final ComparatorX COMPARATOR_X = new ComparatorX();
+    private static final ComparatorY COMPARATOR_Y = new ComparatorY();
+    private static int firstPlace, secondPlace, thirdPlace, fourthPlace;    // indexes of main places for numbers on board
+    private static int threeElementsFirst, threeElementsSecond, threeElementsThird, auxiliaryCoordinateIndex;   // auxiliary variables
+    private static int forTwoElementsRowWithoutChangesFirstPlace, forTwoElementsRowWithoutChangesSecondPlace;   // auxiliary variables
+
+    //  x0  x1  x2   x3
+    //  [2]  [2]  [ ]  [2] y0
+    //  [ ]  [4]  [4]  [ ] y1
+    //  [ ]  [2]  [ ]  [ ] y2
+    //  [2]  [ ]  [3]  [ ] y3
+
+    public static void eachRowProcessing (CoordinateGetter rowGetter,
+                                          CoordinateSetter rowSetter,
+                                          Comparator<Point> comparator,
+                                          CoordinateGetter coordinateGetter,
+                                          CoordinateSetter coordinateSetter) {
+        IntStream.range(0, 4).forEach(rowIndex -> {
+
+            List<Point> certainLineOfGameField = TEMPORARY_LIST.stream()
+                    .filter(point -> rowGetter.getCoordinate(point) == rowIndex)
+                    .sorted(comparator)
+                    .collect(Collectors.toList());
+
+            switch (certainLineOfGameField.size()) {
+                case 1:
+                    oneElementProcessing(certainLineOfGameField.get(0), coordinateGetter, coordinateSetter);
+                    certainLineOfGameField.clear();
+                    break;
+                case 2:
+                    twoElementsProcessing(certainLineOfGameField, rowIndex, rowSetter, coordinateSetter);
+                    break;
+                case 3:
+                    threeElementsProcessing(certainLineOfGameField, coordinateSetter);
+                    break;
+                case 4:
+                    fourElementsProcessing(certainLineOfGameField);
+                    break;
+            }
+        });
+    }
+
+    public static List<Point> moveLeft(List<Point> allPointsList) {
```

Comment:
zdajesz sobie sprawe z tego ze zwracajac po prostu liste, klasa zewnetrzna ktora dostanie ta liste bedzie mogla zmienic stan Twoich obiektow. Bedzie mogla wziasc dowolny punkt i zmienic jego wspolrzedne. Jezeli korzystasz z tej funkcjonalnosci, fajnie ze dziala ale ja bym to przemyslal. Modyfikowanie stanu z roznych miejsc w kodzie predzej czy pozniej doporowadzi do sytuacji w ktorej stan zostanie zmowyfikowany ale nie wiadomo gdzie bo kazdy zmienia to samo i trzyma referencje do tego samego obiektu. 

Zeby tego uniknac mozesz zwrocic kopie takiej listy :)

Reply:
Rozumiem i dokładnie tak robię w innym projekcie z pieniędzmi. Tutaj pozwoliłem sobie zrobić tak bo to tylko gra.